shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -0,0 +1,760 @@
+            ("picking_id.picking_type_id", "in", self.picking_types.ids),
+        ]
+        if lot:
+            lot_domain = [("lot_id", "=", lot.id)]
+            domain = AND([domain, lot_domain])
+        return domain
+
+    def _scan_product__select_move_line(
+        self, product, location, lot=None, packaging=None
+    ):
+        domain = self._scan_product__select_move_line_domain(product, location, lot=lot)
+        move_line = self.env["stock.move.line"].search(domain, limit=1)
+        if move_line:
+            stock = self._actions_for("stock")
+            if self.work.menu.no_prefill_qty:
+                # First, mark move line as picked with qty_done = 0,
+                # so the move wont be split because 0 < qty_done < product_uom_qty
+                stock.mark_move_line_as_picked(move_line, quantity=0)
+                # Then, set the no prefill qty on the move line
+                qty_done = 1
+                if packaging:
+                    qty_done = packaging.qty
+                move_line.qty_done = qty_done
+            else:
+                stock.mark_move_line_as_picked(move_line)
+            return self._response_for_set_quantity(move_line)
+
+    def _scan_product__check_create_move_line(
+        self, product, location, lot=None, packaging=None
+    ):
+        if not self.is_allow_move_create():
+            message = self.msg_store.no_operation_found()
+            return self._response_for_select_product(location, message=message)
+
+    def _scan_product__unreserve_move_line(
+        self, product, location, lot=None, packaging=None
+    ):
+        unreserve = self._actions_for("stock.unreserve")
+        if self.work.menu.allow_unreserve_other_moves:
+            move_lines = self._find_location_move_lines(location, product, lot=lot)
+            response = unreserve.check_unreserve(location, move_lines, product, lot)
+            if response:
+                return response
+            unreserve.unreserve_moves(move_lines, self.picking_types)
+        else:
+            # If we get there then no qty is available, and we are not allowed to unreserve
+            # other moves. No stock available for product.
+            return self._scan_product__no_stock_available(
+                product, location, lot=lot, packaging=packaging
+            )
+
+    def _scan_product__create_move_line(
+        self, product, location, lot=None, packaging=None
+    ):
+        available_quantity = product.with_context(
+            location_id=location.id, lot=lot.id if lot else None
+        ).free_qty
+        is_product_available = (
+            float_compare(
+                available_quantity,
+                packaging.qty if packaging else 1.0,
+                precision_rounding=product.uom_id.rounding,
+            )
+            >= 0
+        )
+        if is_product_available:
+            if packaging:
+                available_quantity = packaging.qty
```

Comment:
why?
you can create the move for the product available quantity. In create move, you will set the quantity done based on the packaging quantity

Reply:
Because the fact that `available_qty >= packaging.qty` when packaging is defined is now handled above.
Or this was wrong in the first place https://github.com/OCA/wms/compare/22a3e5b09df479f30fcd84e51940088f98e92183..9a0c0897de362d793687d87b9524ad938ccafc20#diff-f871450f795713d4ab1061b7117262fdee975730cb8f85f5963d06d339d7e39dL232-R236
cc @sebalix

Reply:
@sebalix 
We discussed with @jbaudoux 
We want to create moves with qty = qty_available, no matter what has been scanned.
This is the qty done that changes when you scan a product/lot/packaging, if `no_prefill_qty` is enabled.
I'll remove this block of code.